lib/tuntap.ml

```diff
@@ -0,0 +1,11 @@
+(*
+ * Copyright (c) Citrix
+ *)
+
+external _tap_open : string -> Unix.file_descr = "stub_tap_open"
+
+let tap_open ifname =
```

Comment:
Would it be useful to provide a `with_tap ifname ~fd:(Unix.file_descr -> 'a)` that also makes sure of closing in a finally block?